### DISCUSSION TITLE
RAK11300 hangs in

### DISCUSSION THREAD

**Discussion by jamcinnes2** [beegee-tokyo/SX126x-Arduino#131] [Q&A] [answered]
Hi Im trying to do a simple Lora sketch on a RAK 11300. I'm using your library and the Earle Philhower RP2040 BSP. It compiles and uploads ok. When it runs it hangs in lora_rak11300_init(). Sounds like you got this working... Can you give me a hint?

```
#include <SX126x-Arduino.h>

void setup()
{

// Initialize Serial for debug output
time_t timeout = millis();
Serial.begin(115200);
while (!Serial)
{
if ((millis() - timeout) < 5000)
{
      delay(100);
      }
      else
      {
          break;
      }
}    
  
Serial.println("=====================================");
Serial.println("LoRap2p Tx Test");
Serial.println("=====================================");

// Initialize LoRa chip.
if (lora_rak11300_init() != 0)
{
   Serial.println("this never prints")
   Serial.println("Failed initialization of SX1262");
}
Serial.println("this never prints either")

}
```

**Comment by beegee-tokyo** [comment, marked as answer]
I didn't use Earl's BSP for a long time. Maybe something has changed. 
![image](https://github.com/user-attachments/assets/9d02d90c-d429-44e8-a51f-61ba8543220d)

It turns out quite difficult to support both Arduino-MBed and Earl's BSP in one code. Might be some defines that are used have changed.

**Comment by jamcinnes2** [comment]
Ok thanks
